pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -189,25 +190,76 @@ func injectApmTelemetryConfig(pod *corev1.Pod) {
 	_ = injectEnv(pod, instrumentationInstallIDEnvVar)
 }
 
-func getAllLibsToInject(registry string) map[string]libInfo {
-	libsToInject := map[string]libInfo{}
+// getLibrariesToInjectForApmInstrumentation returns the list of tracing libraries to inject, when APM Instrumentation is enabled
+// - if apm_config.instrumentation.lib_versions set, returns only tracing libraries from apm_config.instrumentation.lib_versions
+// - if language detection is on and can detect the apps' languages, returns only auto-detected languages
+// - otherwise returns all tracing libraries supported by APM Instrumentation
+func getLibrariesToInjectForApmInstrumentation(pod *corev1.Pod, registry string) ([]libInfo, bool) {
+	autoDetected := false
```

Comment:
nit: autoDetected is true only one case. Code would be easier to read without the variable.

Reply:
I think using true/false  instead of named variable will make it hard to understand what the boolean means without reading docs or following the function calls.

Reply:
Yes, but there is no big difference between current code and using named returned variables. `autoDetected = false` needs to be set at the beginning of the function and update to true when auto-detection is called in either case.